src/affordance/tasks/tools.py

```diff
@@ -25,7 +25,49 @@
 - (str) the output
 - (str) runtime details to be logged
 """
+import pickle
 
+def get_run_dict(run_title):
+    try:
+        with open(f'{run_title}.pickle', 'rb') as handle:
+            run_dict = pickle.load(handle)
+    except FileNotFoundError:
+        run_dict = {}
+        print('run dict not found, creating a new one!')
+    return run_dict
+
+def save_run_dict(run_dict, run_title):
+    with open(f'{run_title}.pickle', 'wb') as handle:
+        pickle.dump(run_dict, handle, protocol=pickle.HIGHEST_PROTOCOL)
+    return
+
+
+def read(key, run_title) -> tuple(str, str):
```

Comment:
The way the code works currently, the first argument is the input text for the command, and the second argument is the output from the previous command. We should probably change the way the tools are called across the board, so they all have three arguments: a `run_title`, a `cmd_input`, and a `prev_output`. Mind adding that change into this PR?